### DISCUSSION TITLE
10007: Unknown Member when trying to remove a reaction on user leave

### DISCUSSION THREAD

**Discussion by AtomXGer24** [discord-jda/JDA#1607] [Questions and Help] [answered]
Hi there,

I got a small issue when trying to remove a reaction when a users leaves the server.

When trying to remove an emoji it works fine with the command.
When trying to remove a custom emote it throws an  
> 10007: Unknown Member ErrorResponseException

However, the reaction get removed when the user leaves.

Is there any way to fix this or should I just ignore the Exception

---
Here is my code
```java
public class UserLeaveListener extends ListenerAdapter {

	// ------------------------------------------------------
	// Listens for user leaving the server
	// ------------------------------------------------------
  @Override	
  public void onGuildMemberRemove(GuildMemberRemoveEvent event) {
    try {
	    // getting the guild from the event
	    Guild guild = event.getGuild();
	    
	    // the ID of the rule accept message			
	    long messageIDRule = 833117996460408843l;
	    
	    // the ID of the role assign message	
	    long messageIDRoleAssign = 833122315075846185l;
	    
	    // get the user who left
	    User user = event.getUser();
	    
	    // getting the custom emotes with the ID	
	    Emote csgo = guild.retrieveEmoteById(831988425023422464l).complete();	
	    Emote pubg = guild.retrieveEmoteById(831988460118081536l).complete();		
	    
	    // getting the text channel in which the rule accept message is placed
	    TextChannel rule = guild.getTextChannelById(831974686966546432l);
	    
	    // getting the text channel in which the role assign message is placed
	    TextChannel selfRole = guild.getTextChannelById(831573505027604480l);
		    
	    // remove the csgo and the pubg emote from the role assign message			
	    selfRole.removeReactionById(messageIDRoleAssign, csgo, user).queue();
	    selfRole.removeReactionById(messageIDRoleAssign, pubg, user).queue();
	    
	    // remove the :white-check-mark: emote from the rule message
	    rule.removeReactionById(messageIDRule, "U+2705", user).queue();
    } catch(ErrorResponseException e) {
	    e.printStackTrace();
    }
  }
}
```

And these are the settings I set and enabled in the main class:
```java
DefaultShardManagerBuilder builder = DefaultShardManagerBuilder.createDefault(myToken);
builder.setStatus(OnlineStatus.ONLINE);
builder.enableIntents(GatewayIntent.GUILD_MEMBERS, GatewayIntent.GUILD_PRESENCES, GatewayIntent.GUILD_MESSAGE_REACTIONS, GatewayIntent.DIRECT_MESSAGE_REACTIONS, GatewayIntent.GUILD_BANS);
builder.setChunkingFilter(ChunkingFilter.ALL);			
builder.setMemberCachePolicy(MemberCachePolicy.ALL);
builder.enableCache(CacheFlag.ACTIVITY);		
builder.enableCache(CacheFlag.CLIENT_STATUS);
builder.enableCache(CacheFlag.ONLINE_STATUS);
builder.enableCache(CacheFlag.MEMBER_OVERRIDES);
builder.addEventListeners(new UserLeaveListener());
builder.build();
```
---

Here is the strack trace:

```
net.dv8tion.jda.api.exceptions.ErrorResponseException: 10007: Unknown Member
net.dv8tion.jda.api.exceptions.ErrorResponseException: 10007: Unknown Member
```

---

These are my settings in the discord dev portal:

![1](https://user-images.githubusercontent.com/45603304/115593739-2c4e6100-a2d5-11eb-9496-b3f31223c1b1.PNG)

I hope this is enough for you to support me.

**Comment by DV8FromTheWorld** [comment, marked as answer]
**Edit**:
User stopped encountering the problem after updating to a newer version of JDA.

**Original Post**:
I haven't had a chance to test this yet but it _looks_ like you're doing everything correctly. It is possible that this is a Discord bug.

Seems possible that discord is telling you `Unknown Member` because their systems are attempting to make sure that the `userId` being provided is a member of the current guild so that they can do permission checks to make sure your bot has permission to remove the reaction of the provided user. When their systems determine that the provided userId _isn't_ in the guild there is probably an gate you're hitting that automatically responds with `Unknown Member`. 

It'll have to test to be sure.

**Reply by AtomXGer24** [reply]
Hey,

thanks for the quick reply.
I´m looking forward for your test result. :)

**Reply by DV8FromTheWorld** [reply]
Another developer was able to create what appeared the same code and it was successful for them. Admittedly, they used a different library and 2 bots instead of a bot managing reactions of a real user, but it has given me pause on my thought about this being a Discord bug.

I haven't had to to write a test case myself yet. Just providing an update.

**Reply by AtomXGer24** [reply]
Ok, thanks for the update.
I´m also trying to get a fix for this all the time, but I wasn´t able to find a solution yet.

**Reply by AtomXGer24** [reply]
Hey there,

I know it´s been a while, but were you able to figure out a solution for this issue?
Even with the newest version, the issue still occures.

**Reply by DV8FromTheWorld** [reply]
Can you produce a minimum reproduction repository so that I can run the code for the test?

**Reply by AtomXGer24** [reply]
Hi,

I just noticed, while creating the project, that after updating the JDA to the latest version, the issue is gone.
I can't reproduce the error anymore. Whatever you did, it works now.

I guess the issue can be closed now.

Nevertheless, thank you for your efforts!

**Reply by DV8FromTheWorld** [reply]
Out of curiosity, what version of JDA were you on before you updated?

**Reply by AtomXGer24** [reply]
I think it was 4.3.0_277.

**Reply by DV8FromTheWorld** [reply]
Thanks!
